inspirehep/modules/records/receivers.py

```diff
@@ -361,3 +361,21 @@ def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
         else:
             record = get_db_record(collection, control_number)
             soft_delete_pidstore_for_record(record.id)
+
+
+@after_record_update.connect
+def check_if_record_has_merged(sender, *args, **kwargs):
```

Comment:
I'll keep it as it's right now because of "race conditions" in SQLalchemy.
I can investigate it more but I don't think that's the proper solution.